hypothesis-python/tests/cover/test_reflection.py

```diff
@@ -758,11 +757,41 @@ def test_cache_key_size_is_bounded():
     f.__code__ = f.__code__.replace(
         co_consts=tuple(c * 1000 if c == "a" else c for c in f.__code__.co_consts)
     )
-    assert len(repr(reflection._lambda_source_key(f))) > 1000
-    assert len(repr(reflection._lambda_source_key(f, bounded_size=True))) < 1000
+    assert len(repr(lambda_sources._function_key(f))) > 1000
+    assert len(repr(lambda_sources._function_key(f, bounded_size=True))) < 1000
 
 
-def test_lambda_source_key_distinguishes_alpha_renames():
+def test_function_key_distinguishes_alpha_renames():
     # these terms are equivalent under the lambda calculus, but their
     # representations are not, so they should be cached differently.
-    assert _lambda_source_key(lambda x: x) != _lambda_source_key(lambda y: y)
+    assert lambda_sources._function_key(lambda x: x) != lambda_sources._function_key(
+        lambda y: y
+    )
+
+
+def test_import():
+    import time as t
+
+    f = lambda: t.ctime()
+    assert get_pretty_function_description(f) == "lambda: t.ctime()"
+
+
+@pytest.mark.parametrize("nop_on_f", [True, False])
+def test_code_normalization(nop_on_f):
+    f = lambda x: x
+    g = lambda x: x
+    h = f if nop_on_f else g
+    assert lambda_sources._function_key(f) == lambda_sources._function_key(g)
+
+    # Append a NOP to one of the bytecodes
+    h.__code__ = h.__code__.replace(co_code=h.__code__.co_code + b"\x09\x00")
+    assert lambda_sources._function_key(f) != lambda_sources._function_key(g)
+
+    # ...and then normalize g to match f (adding or removing a NOP)
+    g.__code__ = lambda_sources._normalize_code(f, g)
+    assert lambda_sources._function_key(f) == lambda_sources._function_key(g)
+
+
+def test_lambda_mimicry_with_arg_defaults():
+    f = lambda x=1, *, y=2: (x, y)
+    assert get_pretty_function_description(f) == "lambda x=1, *, y=2: (x, y)"
```

Comment:
add these test cases for completeness? (hey, if we're going overkill, we might as well go truly overkill!)

```
f = lambda x=1, *, y: (x, y)
f = lambda x, *, y=2: (x, y)
f = lambda x, *, y: (x, y)

f = lambda *, y=2: (x, y)
f = lambda *, y: (x, y)

f = lambda x, /, y=1: (x, y)
f = lambda x, /, y: (x, y)
f = lambda x, /: (x, y)
f = lambda x=1, /, y=2: (x, y)
f = lambda x=1, /: (x, y)
```